packages/app/src/server/crowi/index.js

```diff
@@ -372,7 +372,24 @@ Crowi.prototype.setupPassport = async function() {
 };
 
 Crowi.prototype.setupSearcher = async function() {
-  this.searchService = new SearchService(this);
+  const searchDelegatorLogger = loggerFactory('growi:service:search');
+
+  this.searchService = await new SearchService(this);
+
+  if (this.configManager.getConfig('crowi', 'app:elasticsearchReindexOnBoot')) {
+    searchDelegatorLogger.info('Reindex elasticsearch is running');
+    try {
+      this.searchService.rebuildIndex();
+    }
+    catch (err) {
+      // since rebuildIndex() will force to throw `error` in catch section and `finally` section won't be called,
+      // run normalize indices here anyway
+      this.searchService.normalizeIndices();
+    }
+  }
+  else {
+    searchDelegatorLogger.info('ELASTICSEARCH_REINDEX_ON_BOOT value is false, no reindex on boot');
+  }
```

Comment:
I think it's more appropriate to have SearchService do this reindex work so pls move these new codes to `app/src/server/service.ts`